database/postgresconn/postgresconn.go

```diff
@@ -246,16 +183,21 @@ func (p *Postgres) Unlock() error {
 
 func (p *Postgres) Run(migration io.Reader) error {
 	if p.config.MultiStatementEnabled {
-		var err error
-		if e := multistmt.Parse(migration, multiStmtDelimiter, p.config.MultiStatementMaxSize, func(m []byte) bool {
-			if err = p.runStatement(m); err != nil {
-				return false
+		var currentSchema string
+		if err := p.conn.QueryRowContext(context.Background(), "select current_schema()").Scan(&currentSchema); err != nil {
+			return err
+		}
+		fmt.Printf("current schema %v\n", currentSchema)
+		
+		if err := multistmt.Parse(migration, multiStmtDelimiter, p.config.MultiStatementMaxSize, func(m []byte) error {
+			if err := p.runStatement(m); err != nil {
+				return errors.Wrapf(err, "%s", string(m))
 			}
-			return true
-		}); e != nil {
-			return e
+			return nil
+		}); err != nil {
```

Comment:
I find the combined `if (expr); condition {}` construct hard to follow when split across lines, what do you think about splitting it up?   Is the purpose to narrow the scope of the `err`?

Reply:
agree, I feel like this was pre-existing code in the library, I can change